mmseg/models/losses/inverseform_loss.py

```diff
@@ -0,0 +1,329 @@
+
+    Args:
+    classes(int): number of classes
+    weight(Tensor): weight (Tensor, optional): a manual rescaling weight given
+            to each class. If given, it has to be a Tensor of size `C`.
+            Otherwise, it is treated as if having all ones.
+            Used in nn.NLLLoss.
+    reduction (str, optional): Specifies the reduction to apply to the output:
+            ``'none'`` | ``'mean'`` | ``'sum'``. Default: ``'mean'``.
+            Used in nn.NLLLoss
+    ignore_index(int): Specifies a target value that is ignored
+            and does not contribute to the input gradient, Default: 255.
+    norm(bool): whether use norm in calculating batch classes weights,
+            Default: False.
+    upper_bound(float): The upper bound of weights if calculating weights
+            for every classes. Default: 1.0.
+    fp16(bool): whether to use precision=16 when training, Default: False.
+    edge_weight(float):  weight for balanced cross-entropy loss which
+            is weighted cross-entropy loss,Default: 0.3.
+    inv_weight(float): weight for inverseform loss, Default: 0.3.
+    seg_weight(float): weight for segmentation loss which is corss-entropy
+            loss, Default: 1.
+    att_weight(float): weight for edge_attention, which were not mentioned
+            in paper, Default: 0.1.
+    loss_name (str, optional): Name of the loss item. If you want this loss
+            item to be included into the backward graph, `loss_` must be the
+            prefix of the name. Defaults to 'loss_inverseform'.
+    """
+
+    def __init__(self,
+                 classes,
+                 weight=None,
+                 reduction='mean',
+                 ignore_index=255,
+                 norm=False,
+                 upper_bound=1.0,
+                 fp16=False,
+                 edge_weight=0.3,
+                 inv_weight=0.3,
+                 seg_weight=1,
+                 att_weight=0.1,
+                 loss_name='loss_inverseform'):
+        super().__init__()
+        self.num_classes = classes
+        self.seg_loss = ImageBasedCrossEntropyLoss2d(
+            classes=classes,
+            weight=weight,
+            reduction=reduction,
+            ignore_index=ignore_index,
+            norm=norm,
+            upper_bound=upper_bound,
+            fp16=fp16).cuda()
+        self.inverse_distance = InverseTransform2D()
+        self.edge_weight = edge_weight
+        self.seg_weight = seg_weight
+        self.att_weight = att_weight
+        self.inv_weight = inv_weight
+        self._loss_name = loss_name
+
+    def bce2d(self, input, target):
+        """
+        To calculate weighted cross-entropy of two binary boundary map
+        input and target are both binary boundary map
+        Args:
+        input: pred binary boundary map
+        target: ground truth binary boundary map generated by using Sobel
+                filter and threshold on ground truth segmentation mask
+        """
```

Comment:
We should add type hints. The format of the docstring is not compatible with other documents.